packages/dds/tree/src/feature-libraries/modular-schema/schemaChangeFamily.ts

```diff
@@ -0,0 +1,278 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/common-utils";
+import { ChangeEncoder, ChangeFamily, JsonCompatibleReadOnly, ProgressiveEditBuilder } from "../../change-family";
+import { ChangeRebaser } from "../../rebase";
+import {
+    FieldSchema,
+    GlobalFieldKey,
+    LocalFieldKey,
+    SchemaData,
+    TreeSchema,
+    TreeSchemaIdentifier,
+} from "../../schema-stored";
+import { AnchorSet, Delta, FieldKey } from "../../tree";
+import { brand, Invariant, Multimap } from "../../util";
+import { FieldChangeHandler, FieldChangeMap, FieldChangeset, NodeChangeset } from "./fieldChangeHandler";
+import { FullSchemaPolicy } from "./fieldKind";
+
+export class SchemaChangeFamily implements
+    ChangeFamily<any, NodeChangeset>,
+    ChangeRebaser<NodeChangeset>,
+    ChangeEncoder<NodeChangeset> {
+    _typeCheck?: Invariant<any> | undefined;
+
+    constructor(
+        readonly rootSchema: TreeSchemaIdentifier | undefined,
+        readonly schemaData: SchemaData,
+        readonly schemaPolicy: FullSchemaPolicy,
```

Comment:
I don't think we should be taking in view schema (or any schema) here. I think we should just take in the field-kinds.